packages/api/src/external/commonwell/document/document-query-sandbox.ts

```diff
@@ -60,15 +61,17 @@ export async function sandboxGetDocRefsAndUpsert({
   const entries = patientData.docRefs;
   log(`Got ${entries.length} doc refs`);
 
-  const convertibleDocs = patientData.docRefs
-    .map(entry => {
-      return {
-        ...entry,
-        content: { mimeType: entry.docRef.content?.[0]?.attachment?.contentType },
-      };
-    })
-    .filter(doc => isConvertible(doc.content?.mimeType));
+  const docsWithContent = entries.map(entry => {
+    return {
+      ...entry,
+      content: { mimeType: entry.docRef.content?.[0]?.attachment?.contentType },
+    };
+  });
+
+  const convertibleDocs = docsWithContent.filter(doc => isConvertible(doc.content?.mimeType));
   const convertibleDocCount = convertibleDocs.length;
+  const existingFhirDocs = await getDocuments({ cxId: organization.cxId, patientId: patient.id });
+  const existingTitles = existingFhirDocs.map(d => d.content?.[0]?.attachment?.title);
```

Comment:
This could result in an array filled with `undefined`, right?

Could use `.flatMap()` instead, so it only have actual `string` titles.

Reply:
I don't think flatMap() does that - tried it in a local test and the undefined items stay in the array. But I can chain-filter like `.filter(item => item != undefined)`

Reply:
It does if you use it like so:

```ts
const existingTitles = existingFhirDocs.flatMap(d => d.content?.[0]?.attachment?.title ?? [])
// now the compiler knows existingTitles doesn't have undefined items
```

Unfortunately, `.filter()` doesn't result in an updated type, so the compiler will still think it can contain `undefined` items.

Reply:
woah! that's cool! thank you 🙌